modules/table/src/test/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryMsInvokeTest.java

```diff
@@ -0,0 +1,120 @@
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.apache.ignite.internal.distributionzones.rebalance.RebalanceUtil;
+import org.apache.ignite.internal.hlc.HybridClock;
+import org.apache.ignite.internal.hlc.HybridClockImpl;
+import org.apache.ignite.internal.manager.ComponentContext;
+import org.apache.ignite.internal.metastorage.impl.MetaStorageManagerImpl;
+import org.apache.ignite.internal.metastorage.impl.StandaloneMetaStorageManager;
+import org.apache.ignite.internal.partitiondistribution.Assignment;
+import org.apache.ignite.internal.partitiondistribution.Assignments;
+import org.apache.ignite.internal.replicator.TablePartitionId;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests for disaster recovery meta storage invoke that changes {@link RebalanceUtil#pendingChangeTriggerKey(TablePartitionId)}.
+ */
+public class DisasterRecoveryMsInvokeTest extends BaseIgniteAbstractTest {
+    private static final int partNum = 2;
+    private static final int replicas = 2;
+
+    private static final Set<String> nodes1 = IntStream.of(5).mapToObj(i -> "nodes1_" + i).collect(toSet());
+    private static final Set<String> nodes2 = IntStream.of(5).mapToObj(i -> "nodes2_" + i).collect(toSet());
+
+    private static final Set<Assignment> assignments1 = calculateAssignmentForPartition(nodes1, partNum, replicas);
+    private static final Set<Assignment> assignments2 = calculateAssignmentForPartition(nodes2, partNum, replicas);
+
+    private static final TablePartitionId tablePartitionId = new TablePartitionId(1, 1);
+
+    private static final long expectedPendingChangeTriggerKey = 10L;
+
+    private long assignmentsTimestamp;
+
+    private final HybridClock clock = new HybridClockImpl();
+
+    private MetaStorageManagerImpl metaStorageManager;
+
+    @BeforeEach
+    public void setUp() throws ExecutionException, InterruptedException {
+        metaStorageManager = StandaloneMetaStorageManager.create();
+
+        assertThat(metaStorageManager.startAsync(new ComponentContext()), willCompleteSuccessfully());
+
+        metaStorageManager.deployWatches();
+
+        metaStorageManager.put(pendingChangeTriggerKey(tablePartitionId), longToBytesKeepingOrder(1)).get();
+
+        assignmentsTimestamp = clock.now().longValue();
+    }
+
+    @ParameterizedTest
+    @MethodSource("assignments")
+    public void testPendingChangeTriggerKey(
+            Set<Assignment> currentPending,
+            Set<Assignment> pending
+    ) throws Exception {
+        if (currentPending != null) {
+            metaStorageManager.put(
+                    pendingPartAssignmentsKey(tablePartitionId), Assignments.toBytes(currentPending, assignmentsTimestamp)
+            ).get();
```

Comment:
Usually we use `assertThat(*, willCompleteSuccessfully())`, it includes the timeout. Waiting without timeout is a bad habit.

Reply:
refactored